lib/src/cpp/common/LocatorList.hpp

```diff
@@ -0,0 +1,367 @@
+        const std::string& index);
+
+/**
+ * @brief Parse XML node and print the locator physical port.
+ *        TCP only.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be printed.
+ *
+ * @return std::string locator TCP physical port.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is not an integer.
+ */
+std::string print_physical_port(
+        xercesc::DOMElement*& xml_node,
+        const std::string& index);
+
+/**
+ * @brief Parse XML node and print the locator IP address.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be printed.
+ *
+ * @return std::string locator IP address.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is not an integer.
+ */
+std::string print_address(
+        xercesc::DOMElement*& xml_node,
+        const std::string& index);
+
+/**
+ * @brief Parse XML node and print the locator unique LAN ID.
+ *        TCPv4 only.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be printed.
+ *
+ * @return std::string locator TCPv4 unique LAN ID.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is not an integer.
+ */
+std::string print_unique_lan_id(
+        xercesc::DOMElement*& xml_node,
+        const std::string& index);
+
+/**
+ * @brief Parse XML node and print the locator WAN IPv4 address.
+ *        TCPv4 only.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ * @param[in] index Collection element to be printed.
+ *
+ * @return std::string locator TCPv4 WAN IP address.
+ *
+ * @throw ElementNotFound Exception if the list element does not exist, or the list does not contain
+ *        any element in index position.
+ * @throw BadParameter Exception if the index is not an integer.
+ */
+std::string print_wan_address(
+        xercesc::DOMElement*& xml_node,
+        const std::string& index);
+
+/************************************************************************/
+/* Query functions                                                      */
+/************************************************************************/
+
+/**
+ * @brief Number of locators defined.
+ *
+ * @param[in] xml_node Locator list node where all locators are uploaded.
+ *
+ * @return uint32_t Number of locators in the list.
+ *
+ * @throw ElementNotFound Exception if the list has not been set.
+ */
+uint32_t size(
+        xercesc::DOMElement*& xml_node);
```

Comment:
Again, this is only a `getter`:

```suggestion
        const xercesc::DOMElement& xml_node);
```